test/Resque/Tests/JobTest.php

```diff
@@ -362,4 +362,44 @@ public function testDequeueItemWithiWrongArg()
 		$this->assertEquals(Resque::size($queue), 2);
 	}
 
+	public function testUseFactoryToGetJobInstance()
+    {
+        $payload = array(
+            'class' => Some_Job_Class::class,
+            'args' => null
+        );
+        $job = new Resque_Job('jobs', $payload);
+        $factory = $this->getMock(Resque_Job_FactoryInterface::class);
+        $job->setJobFactory($factory);
+        $testJob = $this->getMock(Resque_JobInterface::class);
+        $factory->expects(self::once())->method('create')->will($this->returnValue($testJob));
+        $instance = $job->getInstance();
+        $this->assertInstanceOf(Resque_JobInterface::class, $instance);
+    }
+
+    public function testDoNotUseFactoryToGetInstance()
+    {
+        $payload = array(
+            'class' => Some_Job_Class::class,
```

Comment:
oh... I see... If we need to keep supporting PHP 5.3/5.4 I can't make use of this... :(

What's your advice? I can easily replace with String. Not ideal, but... Whatever you prefer @chrisboulton